src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractDocumentationCommentService.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Roslyn.Utilities;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.LanguageServices
+{
+    internal abstract class AbstractDocumentationCommentService<
+        TDocumentationCommentTriviaSyntax,
+        TXmlNodeSyntax,
+        TXmlAttributeSyntax,
+        TCrefSyntax,
+        TXmlElementSyntax,
+        TXmlTextSyntax,
+        TXmlEmptyElementSyntax,
+        TXmlCrefAttributeSyntax,
+        TXmlNameAttributeSyntax,
+        TXmlTextAttributeSyntax> : IDocumentationCommentService
+        where TDocumentationCommentTriviaSyntax : SyntaxNode
+        where TXmlNodeSyntax : SyntaxNode
+        where TXmlAttributeSyntax : SyntaxNode
+        where TCrefSyntax : SyntaxNode
+        where TXmlElementSyntax : TXmlNodeSyntax
+        where TXmlTextSyntax : TXmlNodeSyntax
+        where TXmlEmptyElementSyntax : TXmlNodeSyntax
+        where TXmlCrefAttributeSyntax : TXmlAttributeSyntax
+        where TXmlNameAttributeSyntax : TXmlAttributeSyntax
+        where TXmlTextAttributeSyntax : TXmlAttributeSyntax
```

Comment:
which is worse, generics?  or just completely duplicating (bugs and all) all the banner text logic?  IMO, the latter is worse :)